extensions/stats/plugin.h

```diff
@@ -397,20 +409,38 @@ class PluginContext : public Context {
       : Context(id, root),
         upstream_closed_(false),
         downstream_closed_(false),
-        tcp_reported_(false) {}
+        tcp_connection_closed_(false),
+        context_id_(id) {
+    request_info_ = std::make_shared<::Wasm::Common::RequestInfo>();
+  }
 
-  void onLog() override { rootContext()->report(false); };
+  void onLog() override { rootContext()->report(*request_info_, false); };
+
+  FilterStatus onNewConnection() override {
+    request_info_->tcp_connections_opened++;
+    rootContext()->addToTCPRequestQueue(context_id_, request_info_);
+    return FilterStatus::Continue;
+  }
+
+  FilterStatus onDownstreamData(size_t size, bool) override {
```

Comment:
The direction is confusing here. Please add some comment.

Reply:
Done